h/schemas/api/user.py

```diff
@@ -96,3 +96,19 @@ class UpdateUserAPISchema(JSONSchema):
             },
         },
     }
+
+    def validate(self, data):
+        appstruct = super(UpdateUserAPISchema, self).validate(data)
+        appstruct = self._whitelisted_properties_only(appstruct)
+        return appstruct
+
+    def _whitelisted_properties_only(self, appstruct):
```

Comment:
There's an opportunity to factor this out and make it a method available to all schemas, as we're doing it in the group schema as well, but I don't want to take the time right at the moment to generalize it.